awx/main/access.py

```diff
@@ -467,7 +467,21 @@ def can_change(self, obj, data):
         return self.user.is_superuser
 
     def can_delete(self, obj):
-        return self.user.is_superuser
+        if not self.user.is_superuser:
+            return False
+        active_jobs = []
+        active_jobs.extend([dict(type="job", id=o.id)
+                            for o in Job.objects.filter(status__in=ACTIVE_STATES,
+                                                        instance_group=obj)])
+        active_jobs.extend([dict(type="project_update", id=o.id)
+                            for o in ProjectUpdate.objects.filter(status__in=ACTIVE_STATES,
+                                                                  instance_group=obj)])
+        active_jobs.extend([dict(type="inventory_update", id=o.id)
+                            for o in InventoryUpdate.objects.filter(status__in=ACTIVE_STATES,
+                                                                    instance_group=obj)])
```

Comment:
couldn't we just filter on UnifiedJobs? Wouldn't that be easier? I'm sure you can get type from somewhere else.

Reply:
models = [c.__class__.__name__.lower() for c in [Job, ProjectUpdate, InventoryUpdate]]
    active_jobs = UnifiedJob.objects.filter(status__in=ACTIVE_STATES, instance_group=obj, polymorphic_ctype__model__in=models)

Reply:
Do we even need the filter `polymorphic_ctype__model__in=models`? Just leave it out, there's no reason to exclude system jobs, for example.